parser/src/test/java/com/facebook/coresql/parser/sqllogictest/java/SqlLogicTest.java

```diff
@@ -0,0 +1,272 @@
+import static com.facebook.coresql.parser.sqllogictest.java.SqlLogicTestConfig.TEST_FILES_ROOT_PATH;
+import static com.facebook.coresql.parser.sqllogictest.java.SqlLogicTestUtils.createFileForResults;
+import static com.facebook.coresql.parser.sqllogictest.java.SqlLogicTestUtils.validateConfigData;
+import static java.lang.String.format;
+import static java.nio.file.FileVisitResult.CONTINUE;
+
+public final class SqlLogicTest
+{
+    private static SqlLogicTestConfig config;
+    private static Map<String, String> propertiesMap;
+    private static AcceptedRdms rdms;
+
+    private static Set<AcceptedRdms> skipIfSet;
+    private static AcceptedRdms onlyIf;
+    private static BufferedWriter writer;
+    private static FileVisitor<Path> fileVisitor;
+
+    private static int numFilesVisited;
+    private static int numStatementsParsed;
+    private static int numStatementsFound;
+
+    private enum SqlLogicTestError
+    {
+        PARSING_ERROR,
+        UNPARSING_ERROR,
+        UNPARSED_DOES_NOT_MATCH_ERROR
+    }
+
+    protected enum AcceptedRdms
+    {
+        ALL,
+        ORACLE,
+        MYSQL,
+        MSSQL,
+        SQLITE,
+        POSTGRESQL
+    }
+
+    public SqlLogicTest() {}
+
+    public void run()
+            throws IOException
+    {
+        setup();
+        Files.walkFileTree(Paths.get(TEST_FILES_ROOT_PATH), fileVisitor);
+        shutdown();
+    }
+
+    private void setup()
+            throws IOException
+    {
+        getConfigData();
+        validateConfigData(propertiesMap);
+        createFileForResults(RESULTS_PATH);
+        setupMiscVariables();
+    }
+
+    private void getConfigData()
+            throws IOException
+    {
+        config = new SqlLogicTestConfig();
+        propertiesMap = config.getPropertiesMap();
+        rdms = AcceptedRdms.valueOf(propertiesMap.get("rdms").toUpperCase());
+    }
+
+    private void setupMiscVariables()
+            throws IOException
+    {
+        writer = new BufferedWriter(new FileWriter(RESULTS_PATH + "/" + "sqllogictest_results.txt"));
+        fileVisitor = new SqlLogicTestFileVisitor();
+        skipIfSet = new HashSet<>();
+        onlyIf = null;
+    }
+
+    private void shutdown()
+            throws IOException
+    {
+        writer.write(
+                "===============================================\n" +
+                        "RESULTS\n" +
+                        "===============================================\n");
+        writer.write(format("RDMS tested against: %s\n", rdms));
+        writer.write(format("%d statements found, %d statements successfully parsed.\n", numStatementsFound, numStatementsParsed));
+        writer.write(format("%s%% of sqllogictest statements were successfully parsed.",
+                new DecimalFormat("#.##").format(numStatementsParsed / (double) numStatementsFound * 100)));
+        writer.close();
+    }
+
+    private void validateParsingOfSqlStatement(String expr)
```

Comment:
Why is this expr? It should be query?

Reply:
Changed to "statement" in order to be more expressive and match signature.

Done